windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -909,24 +1028,65 @@ function TCompileKeymanWeb.JavaScript_ShiftAsString(fkp: PFILE_KEY; FMnemonic: B
 function TCompileKeymanWeb.JavaScript_Store(pwsz: PWideChar): string;
 var
   ch: DWord;
+  n: Integer;
+  rec: TSentinelRecord;
+const
+  wcsentinel: WideString = #$FFFF;
 begin
-  Result := '"';
-	while pwsz^ <> #0 do
+  n := Pos(wcsentinel, pwsz);
+
+  // Start:  plain text store.  Always use for 9.0-, conditionally for 10.0+.
+  if (n = 0) or not IsKeyboardVersion10OrLater then
   begin
-		if PWord(pwsz)^ = UC_SENTINEL then
+    Result := '"';
+    while pwsz^ <> #0 do
     begin
-      Result := Result + '.'; {TODO: fill in .}
-    end
-    else
-    begin
-      ch := GetSuppChar(pwsz);
-      if ch in [Ord('"'), Ord('\')] then Result := Result + '\';
-      Result := Result + Javascript_String(ch);  // I2242
+      if PWord(pwsz)^ = UC_SENTINEL then
+      begin
+        Result := Result + '.'; {TODO: fill in .}
```

Comment:
Comment needs to change to reflect that UC_SENTINEL values are not supported in stores in KMW<10